models/classes/theme/ThemeServiceAbstract.php

```diff
@@ -216,9 +224,17 @@ protected function getThemeDetailsProviders()
 
     protected function isTaoAsToolEnabled(): bool
     {
-        return $this->getServiceManager()
-            ->getContainer()
-            ->get(FeatureFlagChecker::class)
+        return $this->getFeatureFlagChecker()
             ->isEnabled(FeatureFlagCheckerInterface::FEATURE_FLAG_TAO_AS_A_TOOL);
     }
+
+    protected function getFeatureFlagChecker(): FeatureFlagCheckerInterface
+    {
+        return $this->getContainer()->get(FeatureFlagChecker::class);
+    }
+
+    protected function getContainer(): ContainerInterface
```

Comment:
Same as here https://github.com/oat-sa/tao-core/pull/4029/files#r1618503189

```suggestion
    private function getContainer(): ContainerInterface
```